Fixing 64-bit alignment in structs that use sync.atomic

deepakjois opened the pull request:
This change fixes segmentation faults on Arm v7.

From https://golang.org/pkg/sync/atomic/:

> On both ARM and x86-32, it is the caller's responsibility to arrange
for 64-bit alignment of 64-bit words accessed atomically. The first word
in a variable or in an allocated struct, array, or slice can be relied
upon to be 64-bit aligned.

Fixes #311.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/badger/318)
<!-- Reviewable:end -->

manishrjain commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[transaction.go, line 49 at r1](https://reviewable.io:443/reviews/dgraph-io/badger/318#-KyoVdLe3adagQTFoCtc:-KyoVdLf0j8_c8qmHQTD:b-po9v5c) ([raw file](https://github.com/dgraph-io/badger/blob/1e175e1c05bef27f4f4a3bb69fa84aebb3b8dc6b/transaction.go#L49)):*
> ```Go
> 
> type oracle struct {
> 	curRead   uint64
> ```

Mention what's being protected by mutex.

---

*[skl/skl.go, line 58 at r1](https://reviewable.io:443/reviews/dgraph-io/badger/318#-KyoVphvZiO-lpHBeUpR:-KyoVphvZiO-lpHBeUpS:bhd6sd8) ([raw file](https://github.com/dgraph-io/badger/blob/1e175e1c05bef27f4f4a3bb69fa84aebb3b8dc6b/skl/skl.go#L58)):*
> ```Go
> 	//   value size  : uint16 (bits 32-47)
> 	// 12 bytes are allocated to ensure 8 byte alignment also on 32bit systems.
> 	value1 [12]byte
> ```

What's the 1 suffix for?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/badger/318)*
<!-- Sent from Reviewable.io -->

deepakjois commented:
CI flagged a coupled of tests that were failing. I have fixed them up now.

---

Review status: 2 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[transaction.go, line 49 at r1](https://reviewable.io:443/reviews/dgraph-io/badger/318#-KyoVdLe3adagQTFoCtc:-Kyoed_v1-P0H4aWWa5G:b-896fix) ([raw file](https://github.com/dgraph-io/badger/blob/1e175e1c05bef27f4f4a3bb69fa84aebb3b8dc6b/transaction.go#L49)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

Mention what's being protected by mutex.
</blockquote></details>

Done.

---

*[skl/skl.go, line 58 at r1](https://reviewable.io:443/reviews/dgraph-io/badger/318#-KyoVphvZiO-lpHBeUpR:-Kyoee7olyT5fe9VUfIz:b-l8mtqx) ([raw file](https://github.com/dgraph-io/badger/blob/1e175e1c05bef27f4f4a3bb69fa84aebb3b8dc6b/skl/skl.go#L58)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

What's the 1 suffix for?
</blockquote></details>

Did not notice. Renamed it back to value

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/badger/318#-:-KyoejrfpzwjxZBkAjzx:b-8q2mu0)*
<!-- Sent from Reviewable.io -->

gq0 commented:
> What's the 1 suffix for?

Sorry, that was me to just quickly see where the code breaks to apply there the pointer getter.

manishrjain commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/badger/318#-:-Kyp0ohiQt-Yck7fKlBJ:bnfp4nl)*
<!-- Sent from Reviewable.io -->
